WcaOnRails/app/models/user.rb

```diff
@@ -311,6 +311,10 @@ def teams_where_is_leader
     self.team_members.where(team_leader: true).select(&:current_member?).map!(&:team).uniq
   end
 
+  def current_teams
```

Comment:
Similar to the previous comment, I think we could change this to:

```ruby
   has_many :current_team_members, through: :team_members, -> { where_current }
   has_many :current_teams, through: :current_team_members
```

Along with the following in `TeamMember`:

```ruby
class TeamMember < ActiveRecord::Base
   ...
   def self.where_current
      where("end_date IS NULL OR end_date > ?", Date.today)
   end
end
```